// Copyright 2023 The Yock Authors. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

syntax = "proto3";
option go_package="./;proto";
option java_package = "proto";
option csharp_namespace = "proto";

package Yocki;

service YockInterface {
    // Ping is used to detect whether the connection is available
    rpc Ping (PingRequest) returns (PingResponse);
    rpc Call (CallRequest) returns (CallResponse);
    rpc Info (InfoRequest) returns (InfoResponse);
}

message PingRequest {}

message PingResponse {}

message CallRequest {
    string Fn = 1;
    string Arg = 2;
}

message CallResponse {
    string Buf = 1;
}

message InfoRequest {}

message InfoResponse {}
